scripts/update_csv_from_firebase.py

```diff
@@ -0,0 +1,165 @@
+import time
+
+'''
+This script updates the csv located inside 'street2sat-database-csv' bucket 
+with the associated predictions and changes from the firestore. 
+
+Currently it does not store predictions, only these attributes: 
+
+'input_img': the associated image name in street2sat-uploaded
+'latitude': longitude from the image metadata
+'longitude': longitude from the image metadata
+'being_labeled': if the current image is being labeled or not
+'country': country the image was taken in, retrieved from the folder name 
+'test_set': if the current image is part of the test set TODO
+'time': time from the image metadata
+'focal_length': focal length from the image metadata
+'pixel_height': pixel height from image metadata
+
+
+This csv should be sufficient for dataset changes when going through the 'active learning' 
+cycle for improving the YOLO detection model. 
+
+
+In the future, if predictions are stored here as predictions improve, we can track 
+the performance of street2sat over time using the past predictions. The firebase would
+only store the latest predictions. 
+
+'''
+
+
+
+
+def get_images_already_being_labelled():
+    """Gets images already labelled"""
+    images_already_being_labelled = []
+    csv_names = [blob.name for blob in client.list_blobs('street2sat-gcloud-labeling', prefix="") if blob.name.endswith(".csv")]
+
+    for csv_name in tqdm(csv_names, desc="Get already labelled"):
+        uris = pd.read_csv(f"gs://street2sat-gcloud-labeling/{csv_name}", header=None, sep="\n")[0]
+        images_already_being_labelled += uris.to_list()
+
+    # Ensure there are no duplicates in images already being labelled
+    dupes = [item for item, count in collections.Counter(images_already_being_labelled).items() if count > 1]
+    dupes.remove('0') # An index of 0 was erroneously output in previous csv
+    assert len(dupes) == 0, "Found duplicates in images being labelled. One of the labeling tasks needs to be removed."
+    return set(images_already_being_labelled)
+
+
+
+
+# Initialize connections to cloud storage
+client = storage.Client()
+db = firestore.Client()
+coll = db.collection("street2sat")
+
+# csv storage bucket
+uploaded_bucket = client.get_bucket('street2sat-database-csv')
+
+database_info = pd.read_csv('gs://street2sat-database-csv/database-info.csv')
+
+test_set_info = pd.read_csv('gs://street2sat-database-csv/test-set.csv')
+
+# used to see if image is in test set or not 
+test_set_imgs = set(test_set_info['input_img'])
+
+# to upadate any images from not being labaled to being labeled 
+already_labeled = get_images_already_being_labelled()
+
+
+lat = []
+lon = []
+name = []
+being_labeled = []
+country = []
+test_set = []
+time_d = []
+focal_length = []
+pixel_height = []
+dup_checker = set()
+i = 0
+
+# query limit amount bc queries can't live for more than 30 seconds 
+limit_amount = 10000
+
+test_num = 0
+nones_detected = 0
+
+docs = coll.order_by('input_img').limit(limit_amount).stream()
```

Comment:
This is not at all necessary, however, I think it might be cleaner to put all the data into a data frame right away and do operations directly on the data frame:

For example
```
all_data = []
for image in docs:
    all_data.append( image.to_dict() )

df = pd.DataFrame(all_data)

# Remove coords with none, etc 
```
Up to you if you want to take a look at this